outputs/kinesis/kinesis_output.go

```diff
@@ -0,0 +1,155 @@
+package kinesis_output
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"sync/atomic"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+
+	"github.com/influxdb/influxdb/client/v2"
+	"github.com/influxdb/telegraf/outputs"
+)
+
+type KinesisOutput struct {
+	Region       string `toml:"region"`
+	StreamName   string `toml:"streamname"`
+	PartitionKey string `toml:"partitionkey"`
+	Format       string `toml:"format"`
+	Debug        bool   `toml:"debug"`
+	svc          *kinesis.Kinesis
+}
+
+var sampleConfig = `
+  # Amazon REGION of kinesis endpoint.
+  # Most AWS services have a region specific endpoint this will be used by
+  # telegraf to output data.
+  # Authentication is provided by an IAMS role, SharedCredentials or Environment
+  # Variables.
+  region = "ap-southeast-2"
+  # Kinesis StreamName must exist prior to starting telegraf.
+  streamname = "StreamName"
+  # PartitionKey as used for sharding data.
+  partitionkey = "PartitionKey"
```

Comment:
The PartitionKey is a required field in the [API](http://docs.aws.amazon.com/kinesis/latest/APIReference/API_PutRecord.html). I have put it in there (rather than having a default) so that people are aware that the field is mandatory and that it should be set to something that allows them to designate a shard for any given hosts. 

Is this bad practise?